### DISCUSSION TITLE
Den of eveill loop problem

### DISCUSSION THREAD

**Discussion by knewstimek** [isid0re/SoloLeveling#295] [Ideas]
Due to the loop below, there is a bug where you cannot complete the Den of Evil quest and exit. I think the code needs to be fixed.
**_//while (me.area === 8) {
	//	Attack.clearLevel();
	//}_**

/*
*	@filename	den.js
*	@author		isid0re
*	@desc		den quest
*/

function den () {
	print('ÿc9SoloLevelingÿc0: starting den');
	me.overhead("den");

	//콜드플레인 웨이없으면
	if (!Pather.checkWP(3)) {
		Pather.moveToExit(2, true); //블러드무어로 이동
		Pather.moveToExit(8, false, true); //덴오브이블로 이동하되 들어가지는 않음
		Pather.makePortal(); //포탈 연다
		Pather.getWP(3); //콜드플레인 웨이 찍으러간다
		Attack.clear(50); //콜드플레인 주변 사냥
		Pather.useWaypoint(1); //웨이포인트 통해 로그캠프로 이동
	}

	Town.doChores(); //마을에서 할거좀 한다

	//덴오브 이블앞에 열어놓은 포탈 있으면 사용
	if (!Pather.usePortal(2, me.name)) {
		Pather.moveToExit(2, true); //포탈 없으면 그냥 블러드무어로 뛰어
	}

	Precast.doPrecast(true);
	Attack.clear(50); //블러드 무어 사냥
	if(!Pather.moveToExit(8, true)) //사냥하면서 덴오브이블로 이동
		print("ÿc9SoloLevelingÿc0: Failed to move to Den of eveil");
	//현재 맵이 덴오브이블이라면
	//while (me.area === 8) {
	//	Attack.clearLevel();
	//}_

	if(me.area === 8)
		Attack.clearLevel();

	return true;
}

**Comment by isid0re** [comment]
Added a check for corpsefire immunity which prevented the bot from exiting.

**Reply by mirakhany** [reply]
Hi. I have to report one thing.
This is very rare.
When my sorceress(Leveling Build) encountered baal's 5th wave monster who has fire and cold immune, she never stop teleporting.
please check this issue.

**Reply by tyson-dev** [reply]
This is a thing! This is when she tries to skip Lister just to find the next Monster in her way being Lister. Not sure what's best solution here, maybe telestomping or do a custom static attack on Lister if immune to "cold and fire".

like in standard kolbot there's custom attacks but she should really only do it when immun to all other attacks.
	Config.CustomAttack = {
		//"Monster Name": [-1, -1]
		"Lister": ["static or whatever", -1],

**Reply by knewstimek** [reply]
Not because of immune. My character was playing normal diff.
Sorc and Nec have same problem. i didn't check other classes.

To solve this, simply remove the while() in den.js.

Even though the quest is completed, due to 'while(me.area === 8)' Character wanders around Den of Evil without performing the next action.
